pseudocode-defns/pseudocode_defns.tex

```diff
@@ -0,0 +1,490 @@
+
+\newcommand{\inRust}[2]{See \url{#2}.}
+\newcommand{\inOpenDPRust}[2]{In OpenDP (Rust), this is called \texttt{#1}. See \url{#2}.}
+\newcommand{\inPython}[2]{In Python, this is called \texttt{#1}. See \url{#2}.}
+\newcommand{\questionc}[1]{\textcolor{red}{\textbf{Question:} #1}}
+\newcommand{\questionr}[1]{\textcolor{blue}{\textbf{Question for reviewers:}}\textcolor{red}{~#1}}
+\newcommand{\T}{\texttt{T}}
+\newcommand{\data}{\texttt{data}}
+\newcommand{\todo}{{\textcolor{red}{TODO: }}}
+\newcommand{\todonei}{{\textcolor{red}{TODO (future -- not enough info yet): }}}
+\newcommand{\iffText}{\text{if and only if}}
+
+
+\title{List of definitions used in the pseudocode}
+\author{S\'ilvia Casacuberta, Grace Tian, and Connor Wagaman}
+\date{June--July 2021}
+
+\begin{document}
+
+\maketitle
+
+We use the following guideline: if a term appears in the preconditions \& pseudocode section, then this term is defined in the ``List of definitions used in the pseudocode" document. Otherwise, it appears in the ``List of definitions used in the proofs" document. 
+
+In both cases, we maintain the terms in alphabetical order within each section. ``TODOs'' should be included at the end of the corresponding section. On the other hand, ``TODOs'' which better specify an already-defined term should be included immediately following the definition of that term. Examples should never be part of the definition, but we encourage their use right after the definition of a term.
+
+We also recommend linking to the Rust Standard Library in the cases where the term is already defined there.
+
+\tableofcontents
+
+\section{Types}
+% any types should be listed here
+
+\begin{definition}[\texttt{bool}]
+    The type \texttt{bool} represents a value, which can only be either \texttt{True} or \texttt{False}. If you \texttt{cast} a \texttt{bool} into an integer, \texttt{True} will be \texttt{1} and \texttt{False} will be \texttt{0}.
+\end{definition}
+
+\begin{definition}[\texttt{::Carrier}]
+\texttt{SomeDomain::Carrier} is the type of a member in \texttt{SomeDomain}, where \texttt{SomeDomain} is a domain. 
+
+For example, \texttt{AllDomain(T)::Carrier} is equivalent to $\T$.
+\end{definition}
+
+\begin{definition}[\texttt{f32}]$\sfi$
+    \texttt{f32} is the Rust 32-bit floating point type. \inRust{f32}{https://doc.rust-lang.org/std/primitive.f32.html}
+\end{definition}
+
+\begin{definition}[\texttt{f64}]$\sfi$
+    \texttt{f64} is the Rust 64-bit floating point type. \inRust{f64}{https://doc.rust-lang.org/std/primitive.f64.html} 
+\end{definition}
+
+\todonei{Add / pointers to  ``binary64” type defined in IEEE 754-2008.}
+
+\begin{definition}[\texttt{IntDistance}]
+\texttt{IntDistance} is equivalent to \texttt{u32}.
+\end{definition}
+
+\begin{definition}[\texttt{null}]
+    Represents a missing or unknown value. It is not equivalent to any other value, and is counted as an element when calculating symmetric distance.
+\end{definition}
```

Comment:
Is null an element of all types?  That is, can a u32 variable be null, an f64 variable be null, etc.?

Reply:
This definition was misplaced because `null` is not a type. It is now rolled into the definition of the trait `InherentNull`.

Reply:
Got it, thanks. I put some comments on the definition of InherentNull.